src/background_thread.c

```diff
@@ -857,12 +865,12 @@ background_thread_boot1(tsdn_t *tsdn) {
 	}
 
 	background_thread_info = (background_thread_info_t *)base_alloc(tsdn,
-	    b0get(), ncpus * sizeof(background_thread_info_t), CACHELINE);
+	    b0get(), MAX_BACKGROUND_THREAD_LIMIT * sizeof(background_thread_info_t), CACHELINE);
```

Comment:
sizeof(background_thread_info_t) is actually quite a bit (256) and when times 4096 it's 1MB. This overhead becomes real when we use huge pages for metadata (metadata_thp is enabled by default in our setup). How about we don't allow changing the number of threads on the fly?

Reply:
I'd like to at least be allowed to dynamically set it lower to make it easier to test, I'll make this opt_max_background_threads (default to ncpus), since it shouldn't change after startup

Reply:
Set it lower sounds good.